Add support for different environments

migore opened the pull request:
Use MIX_ENV variable to select the build configuration, if it's not set the default is `prod`.

I didn't add anything to the docs because the usage is the same as any other mix task, just set the MIX_ENV. Let me know if you think it's necessary and I'll add. :)

Relates to #13

migore committed:
Add support for different environments

asaaki commented on spec/edib/defaults_spec.exs:
```diff
@@ -19,6 +19,18 @@ defmodule EDIBDefaultsSpec do
 
+    describe "environment/0" do
+      it "returns prod as default environment" do
+        System.delete_env("MIX_ENV")
```
We probably should mock the call to `System.get_env/1` instead, so to avoid possible conflicts if running the specs concurrently.

Something like the following should do the job:

``` elixir
before do
  allow(SomeModule) |> to(accept :get_env, fn "MIX_ENV" -> nil end) # for delete
  allow(SomeModule) |> to(accept :get_env, fn "MIX_ENV" -> "staging" end) # for different values
end
```

You would need to wrap both cases in different `describe` / `context` blocks.

Could you try this and adjust the PR if it's working?

asaaki commented on spec/edib/defaults_spec.exs:
```diff
@@ -19,6 +19,18 @@ defmodule EDIBDefaultsSpec do
+      end
+
+      it "returns MIX_ENV as the environment" do
+        System.put_env("MIX_ENV", "staging")
```
See above.

asaaki commented:
It's funny how I forgot that it is actually possible to override the default MIX_ENV in the container run. 😅 

Regarding documentation:
With this change it is possible to break the standard expectation of users if they put a MIX_ENV in front of their commands on a regular basis. A note would be greatful to avoid confusion. (Along the line like "mix edib will pass through the MIX_ENV you used to call the task with." — haven't thought about the exact copy yet.)

migore committed:
Mock the system module in tests

migore committed:
Document about different environments.

If the user was calling the command with and environemnt variable set
EDIB behavior will chnage, in order to inform the user about what's
happening this information is being added to the readme file.

migore commented:
Great suggestion about the mocking in the tests, it worked! :)

Added some notes on the usage section and a warning saying that the default environment is `prod`.

asaaki commented:
Thank you for your contribution! 🙇
